chatbot/SessionState.js

```diff
@@ -23,6 +15,14 @@ class SessionState {
         this.incidentType = incidentType
         this.notes = notes
         this.fallBackAlertTwilioStatus = fallBackAlertTwilioStatus
+        this.incidentCategoryMessage = 'Now that you have responded, please reply with the number that best describes the incident:\n0 - accidental\n1 - safer use\n2 - unsafe guest\n3 - overdose\n4 - other'
+        this.allowedIncidentTypes = {
+            '0': 'Accidental',
+            '1': 'Safer Use',
+            '2': 'Unsafe Guest',
+            '3': 'Overdose',
+            '4': 'Other'
+        }
     }
 
     advanceSession(messageText) {
```

Comment:
1. Might be clearer to deal with the validity of the message upfront.

```
advanceSession(messageText) {
   ...
   case STATES.WAITING_FOR_CATEGORY:
      try {
         const incidentType = parseIncidentType(messageText)
         this.state = STATES.WAITING_FOR_DETAILS
         return 'Thank you. If you like, you can reply with any further details about the incident.'
      } catch (err) {
         return "'Sorry, the incident type wasn\'t recognized. Please try again."
      }
  ...
}

parseIncidentType(text) {
   // throw if unrecognised
}
```

2. Echo the invalid message to help the user diagnose the problem.

e.g. something like `Sorry, I didn't understand "accdental". Please try again. Valid options are Accidental, Safer Use, Unsafe Guest, Overdose, Other`

3. Default clause
Response is "Unfortunately, we have encountered an error in our system and will deal with it shortly.".

How would we know there has been an error? Recommend either explicitly doing whatever needs to happen to notify us here (e.g. log to Sentry) or propagating the error, e.g.

```
safeAdvanceSession(messageText) {
  try {
    const responseText = advanceSession(messageText)
    safeSendResponse(responseText)
  } catch (error) {
     safeSendResponse("Unfortunately, we have encountered an error in our system and will deal with it shortly.")
     throw error
  }

 // caught and reported by top-level Sentry handler
```

or something to that effect